sdk/storage/CMakeLists.txt

```diff
@@ -8,174 +8,31 @@ set(CMAKE_CXX_STANDARD 14)
-    inc/blobs/blob.hpp
-    inc/blobs/blob_service_client.hpp
-    inc/blobs/blob_container_client.hpp
-    inc/blobs/blob_client.hpp
-    inc/blobs/block_blob_client.hpp
-    inc/blobs/page_blob_client.hpp
-    inc/blobs/append_blob_client.hpp
-    inc/blobs/blob_options.hpp
-    inc/blobs/blob_responses.hpp
-    inc/blobs/blob_sas_builder.hpp
-    inc/blobs/protocol/blob_rest_client.hpp
-)
-
-set (AZURE_STORAGE_BLOB_SOURCE
-    src/blobs/blob_service_client.cpp
-    src/blobs/blob_container_client.cpp
-    src/blobs/blob_client.cpp
-    src/blobs/block_blob_client.cpp
-    src/blobs/page_blob_client.cpp
-    src/blobs/append_blob_client.cpp
-    src/blobs/blob_sas_builder.cpp
-)
-
-add_library(azure-storage-blob ${AZURE_STORAGE_BLOB_HEADER} ${AZURE_STORAGE_BLOB_SOURCE})
-target_link_libraries(azure-storage-blob azure-storage-common)
-
-set (AZURE_STORAGE_DATALAKE_HEADER
-    inc/datalake/datalake.hpp
-    inc/datalake/protocol/datalake_rest_client.hpp
-    inc/datalake/datalake_responses.hpp
-    inc/datalake/datalake_options.hpp
-    inc/datalake/datalake_utilities.hpp
-    inc/datalake/datalake_service_client.hpp
-    inc/datalake/datalake_file_system_client.hpp
-    inc/datalake/datalake_path_client.hpp
-    inc/datalake/datalake_file_client.hpp
-    inc/datalake/datalake_directory_client.hpp
-)
-
-set (AZURE_STORAGE_DATALAKE_SOURCE
-    src/datalake/datalake_service_client.cpp
-    src/datalake/datalake_file_system_client.cpp
-    src/datalake/datalake_path_client.cpp
-    src/datalake/datalake_file_client.cpp
-    src/datalake/datalake_directory_client.cpp
-    src/datalake/datalake_utilities.cpp
-    src/datalake/datalake_responses.cpp
-)
-
-add_library(azure-storage-file-datalake ${AZURE_STORAGE_DATALAKE_HEADER} ${AZURE_STORAGE_DATALAKE_SOURCE})
-target_link_libraries(azure-storage-file-datalake azure-storage-blob)
-
-set (AZURE_STORAGE_SHARES_HEADER
-    inc/shares/shares.hpp
-    inc/shares/protocol/share_rest_client.hpp
-    inc/shares/share_constants.hpp
-    inc/shares/share_file_attribute.hpp
-    inc/shares/share_options.hpp
-    inc/shares/share_responses.hpp
-    inc/shares/share_service_client.hpp
-    inc/shares/share_client.hpp
-    inc/shares/share_directory_client.hpp
-    inc/shares/share_file_client.hpp
-)
-
-set (AZURE_STORAGE_SHARES_SOURCE
-    src/shares/share_service_client.cpp
-    src/shares/share_client.cpp
-    src/shares/share_directory_client.cpp
-    src/shares/share_file_client.cpp
-)
-
-add_library(azure-storage-file-share ${AZURE_STORAGE_SHARES_HEADER} ${AZURE_STORAGE_SHARES_SOURCE})
-target_link_libraries(azure-storage-file-share azure-storage-common)
-
-add_library(azure::storage::common ALIAS azure-storage-common)
-add_library(azure::storage::blob ALIAS azure-storage-blob)
-add_library(azure::storage::file::datalake ALIAS azure-storage-file-datalake)
-add_library(azure::storage::file::share ALIAS azure-storage-file-share)
 
+option(BUILD_STORAGE_SAMPLES "Build storage sample codes" ON)
```

Comment:
Nit, place the options at the top of the CMakeList so they are easy to find

Reply:
fixed